Geography/Scottish Postcode Directory/2_Create Postcode Lookup Files.R

```diff
@@ -123,50 +135,83 @@ SPD <- read.csv(file.path(data_filepath, "SingleRecord.csv"), stringsAsFactors =
 ### 3 - Change Variable Formats ----
 
 # Change Date Variables Into Correct Format
-SPD <- SPD %>%
-  mutate(Date_Of_Introduction = as.Date(Date_Of_Introduction, "%d/%m/%Y")) %>% 
-  mutate(Date_Of_Deletion = as.Date(Date_Of_Deletion, "%d/%m/%Y"))
+
+SPD %<>%
+  mutate(Date_Of_Introduction = as.Date(Date_Of_Introduction, "%d/%m/%Y"), 
+         Date_Of_Deletion = as.Date(Date_Of_Deletion, "%d/%m/%Y"))
+
 
 # Add leading zeroes to some columns
-# If this is missed the file won't show as a 100% comparison with the SPSS version for quality assurance check
-SPD$Locality_2001 <- str_pad(SPD$Locality_2001, 6, pad = "0")
-SPD$Settlement_2001 <- str_pad(SPD$Settlement_2001, 3, pad = "0")
-SPD$Locality_1991 <- str_pad(SPD$Locality_1991, 3, pad = "0")
-SPD$Islands_2016 <- str_pad(SPD$Islands_2016, 3, pad = "0")
+# If this is missed the file won't show as a 100% comparison with the SPSS 
+# version for quality assurance check
+
+SPD %<>% 
+  mutate(Locality_2001 = str_pad(SPD$Locality_2001, 6, pad = "0"), 
+         Settlement_2001 = str_pad(SPD$Settlement_2001, 3, pad = "0"), 
+         Locality_1991 = str_pad(SPD$Locality_1991, 3, pad = "0"), 
+         Islands_2016 = str_pad(SPD$Islands_2016, 3, pad = "0"))
 
 # Create 7 character postcode variable
-SPD <- SPD %>%
+
+SPD %<>% 
   mutate(pc7 = ifelse(nchar(pc8) == 6, gsub(" ","  ",pc8),
                ifelse(nchar(pc8) == 7, pc8,
-               ifelse(nchar(pc8) == 8, gsub(" ","",pc8),NA)))) %>%
+               ifelse(nchar(pc8) == 8, gsub(" ", "", pc8), NA)))) %>%
   arrange(pc7, desc(Date_Of_Introduction), Date_Of_Deletion)
 
-# Add columns for HB2019Name, HSCP2019Name, CA2019Name, DataZone2011Name and IntZone2011Name
+
+# Add columns for HB2019Name, HSCP2019Name, CA2019Name, DataZone2011Name and 
+# IntZone2011Name
 
 # Use the Geography Codes and Names open data file to get the names
-# read_csv producing a timeout error when Calum tried to run it so best using read.csv
+# read_csv producing a timeout error when Calum tried to run it so best 
+# using read.csv
 
-geo_names <- read.csv(file = "https://www.opendata.nhs.scot/dataset/9f942fdb-e59e-44f5-b534-d6e17229cc7b/resource/395476ab-0720-4740-be07-ff4467141352/download/geography_codes_and_labels_dz2011_01042019.csv") %>% 
+geo_names <- read.csv(paste0("https://www.opendata.nhs.scot/dataset/", 
```

Comment:
We can use API call to retrieve data from CKAN, so that when the data get updated on CKAN we don't need to manually update the URL. You can have a look at the API example in the Open-Data repository.